tdm-be/src/controller/data-enrichment.ts

```diff
@@ -62,28 +61,80 @@ const router = express.Router();
  *                 $ref: '#/components/schemas/Request'
  */
 
-//Route to access all enrichments in the istex api due to config set
-router.get('/list', async (req, res) => {
-    const results = await Promise.all(
-        singleton.getConfig().enrichments?.map((enrichment) => axios.get(enrichment.url)),
-    );
+type PathParameters = {
+    name: string;
+};
+
+type Path = {
+    post: {
+        summary: string;
+        description: string;
+        tags?: string[];
+        parameters?: PathParameters[];
+    };
+};
+
+type ServerVariable = {
+    description: string;
+    default: string;
+    enum?: string[];
+};
+
+type Server = {
+    url: string;
+    description: string;
+    variables?: Record<string, ServerVariable>;
+};
+
+type Enrichment = {
+    servers: Server[];
+    paths: Record<string, Path>;
+};
 
-    const requests = results.flatMap((res) => {
-        const tags = singleton.getConfig().enrichments?.find((enrichment) => enrichment.url === res.config.url)?.tags;
-        return Object.entries(res.data.paths)
-            .filter((path: any) =>
-                tags?.some((t) => path[1].post?.tags?.includes(t.name) && t.excluded?.indexOf(path[0]) === -1),
-            )
-            .map<Request>((path: any) => {
-                return {
-                    label: `${path[1].post.summary} (${path[0]})`,
-                    description: path[1].post.description,
-                    url: `${res.data.servers[0].variables.scheme.default}://${res.data.servers[0].variables.hostname.default}${path[0]}`,
-                    parameters: path[1].post?.parameters?.map((param: any) => param.name),
-                };
-            });
+const getEnrichments = async () => {
+    const enrichments = [];
+
+    for (const enrichment of dynamicConfig.getConfig().enrichments) {
+        const ws = await axios.get<Enrichment>(enrichment.url);
+        const paths = ws.data.paths;
+        const servers = ws.data.servers;
+        const tags = enrichment.tags.flatMap((tag) => tag.name);
+        const excluded = enrichment.tags.flatMap((tag) => tag.excluded).filter((tag) => !!tag) as unknown as string[];
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought (non-blocking):** ‏`filter((tag) => !!tag)` equivalent to `filter(Boolean)`?

Reply:
I think so, but I prefer to make the filter coparator explicit.

Reply:
I don't find `!!tag` explicit.
An explicit way would be `(tag) => Boolean(tag)`, which is equivalent to `Boolean`.